third_party/move/move-model/bytecode/src/fat_loop.rs

```diff
@@ -450,20 +450,57 @@ impl FatLoopBuilder {
         cfg: &StacklessControlFlowGraph,
         sub_loops: &[NaturalLoop<BlockId>],
     ) -> Vec<Vec<Bytecode>> {
-        sub_loops
+        let mut id_label_map = BTreeMap::new();
+        let blocks: Vec<(BlockId, Vec<Bytecode>)> = sub_loops
             .iter()
             .flat_map(|l| l.loop_body.iter())
             .map(|block_id| match cfg.content(*block_id) {
                 BlockContent::Dummy => {
                     panic!("A loop body should never contain a dummy block")
                 },
                 BlockContent::Basic { lower, upper } => {
+                    if let Some(lbl) = code[*lower as usize].get_label_inner_opt() {
+                        id_label_map.insert(*block_id, lbl);
+                    }
                     let block: Vec<_> = (*lower..=*upper)
                         .map(|i| code.get(i as usize).unwrap().clone())
                         .collect();
-                    block
+                    (*block_id, block)
                 },
             })
-            .collect()
+            .collect();
+        let mut results = vec![];
+        for (i, (block_id, block)) in blocks.iter().enumerate() {
+            // if this block has one successor and the last bc of this block is not a branch
```

Comment:
"If this is a fall-through block (block has one successor...)" we need to remove it by ...
The fatloop algorithm doesn't support fallthrough.